modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/apps-search-results/ddm-template.ftl

```diff
@@ -79,39 +79,73 @@
 </#if>
 
 <#assign
+	channelId = ""
+	channels = restClient.get("/headless-commerce-delivery-catalog/v1.0/channels")
+	developerName = ""
 	pageSize = pageSize?has_content?then(pageSize, 15)
 	page = page?has_content?then(page, 1)
 	taxonomyVocabularyName = "Marketplace Product Type"
 	categoryName = "App"
 	taxonomyVocabulary = restClient.get("/headless-admin-taxonomy/v1.0/sites/${themeDisplay.getCompanyGroupId()}/taxonomy-vocabularies?fields=id&filter=name eq '${taxonomyVocabularyName}'").items
 	vocabularyCategory = restClient.get("/headless-admin-taxonomy/v1.0/taxonomy-vocabularies/${taxonomyVocabulary[0].id}/taxonomy-categories?fields=id&filter=name eq '${categoryName}'").items
-	productsList = restClient.get("/headless-commerce-admin-catalog/v1.0/products?filter=categoryIds/any(params:params eq '${vocabularyCategory[0].id}')&pageSize=" + pageSize + "&page=" + page)
+	productsList = restClient.get("/headless-commerce-admin-catalog/v1.0/products?filter=categoryIds/any(params:params eq '${vocabularyCategory[0].id}')&pageSize=" + pageSize + "&page=" + page).items
+	priceModel = ""
 	numberFilteredProducts = 0
 	filterCategoriesByUrlParams = getFilterByUrlParams(siteURL)
 />
 
-<#if filterCategoriesByUrlParams?has_content>
-	<#assign
-		productsList = restClient.get("/headless-commerce-admin-catalog/v1.0/products?filter=categoryIds/any(params:${filterCategoriesByUrlParams} and (params eq '${vocabularyCategory[0].id}'))&pageSize=" + pageSize + "&page=" + page)
-	/>
-</#if>
+<#list channels.items as channel>
+	<#if channel.name == "Marketplace Channel">
+		<#assign channelId = channel.id />
+	</#if>
+</#list>
+
+<#list entries as curCPCatalogEntry>
+	<#assign cpDefinitionId = curCPCatalogEntry.getCPDefinitionId() />
+
+	<#if (curCPCatalogEntry.getCProductId())??>
+		<#assign specifications = restClient.get("/headless-commerce-delivery-catalog/v1.0/channels/" + channelId + "/products/" + curCPCatalogEntry.getCProductId() + "/product-specifications") />
```

Comment:
Hey, Elias! Do we really need to use the commerce delivery catalog api to get the product specifications? Is there any difference from the admin catalog that we were already using? Because if you get the same informations, we don't need all requests to get channelId

Reply:
A definição da requisição de channel é necessárias pois o especification requer um parametro de channelId. Não identifiquei no código outra forma de buscar essa informação. Escolhi o end point delivery pois é o mais indicado pra qualquer instância do MKP. Você pensa em uma forma melhor pra fazer isso?